src/ResourceManager/Kusto/Commands.Kusto/Utilities/KustoUtils.cs

```diff
@@ -0,0 +1,87 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Azure.Commands.Kusto.Utilities
+{
+    class KustoUtils
+    {
+        public static void GetResourceGroupNameAndClusterNameFromClusterId(
+            string clusterId,
+            out string resourceGroupName,
+            out string clusterName)
+        {
+            string[] tokens = ParseClusterResourceId(clusterId);
+            resourceGroupName = tokens[3];
+            clusterName = tokens[7];
+        }
+
+        public static void GetResourceGroupNameClusterNameAndDatabaseNameFromDatabaseId(
+            string databaseId,
+            out string resourceGroupName,
+            out string clusterName,
+            out string databaseName)
+        {
+            string[] tokens = ParseDatabaseResourceId(databaseId);
+            resourceGroupName = tokens[3];
+            clusterName = tokens[7];
+            databaseName = tokens[9];
+        }
+
+        public static void GetResourceGroupNameFromClusterId(
+            string clusterId,
+            out string resourceGroupName)
+        {
+            string[] tokens = ParseClusterResourceId(clusterId);
+            resourceGroupName = tokens[3];
+        }
+
+        private static string[] ParseClusterResourceId(string clusterId)
```

Comment:
Please use this instead: https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs